image_tools/src/cam2image.cpp

```diff
@@ -91,36 +88,23 @@ int main(int argc, char * argv[])
   // even when executed simultaneously within a launch file.
   setvbuf(stdout, NULL, _IONBF, BUFSIZ);
 
+  // Set the parameters of the quality of service profile. Initialize as the default profile
+  // and set the QoS parameters specified on the command line.
+  rmw_qos_profile_t custom_camera_qos_profile = rmw_qos_profile_default;
+
   // Configure demo parameters with command line options.
-  bool success = parse_command_options(
-    argc, argv, &depth, &reliability_policy, &history_policy, &show_camera, &freq, &width, &height,
-    &burger_mode, &topic);
-  if (!success) {
+  if (!parse_command_options(
+      argc, argv, &(custom_camera_qos_profile.depth), &(custom_camera_qos_profile.reliability),
+      &(custom_camera_qos_profile.history), &show_camera, &freq, &width, &height,
+      &burger_mode, &topic))
+  {
     return 0;
   }
 
   // Initialize a ROS 2 node to publish images read from the OpenCV interface to the camera.
   auto node = rclcpp::Node::make_shared("cam2image");
   rclcpp::Logger node_logger = node->get_logger();
 
-  // Set the parameters of the quality of service profile. Initialize as the default profile
-  // and set the QoS parameters specified on the command line.
```

Comment:
I found these comments valuable even if they come at the cost of a few local variables.
Do you mind keeping `reliability_policy` and friends variables but initialize them with the default profile values instead of the previously hard-coded ones?

Reply:
I did basically this, but just re-used the fields of the structure.  The code looks slightly weird this way, but at least has the documentation that you liked.  I could re-introduce the local variables, but it seems like more of a way to make a mistake than anything else.  Let me know what you think.